lib/rules/max-statements-per-line.js

```diff
@@ -35,11 +35,11 @@ module.exports = {
         const sourceCode = context.getSourceCode(),
             options = context.options[0] || {},
             maxStatementsPerLine = typeof options.max !== "undefined" ? options.max : 1,
-            message = "This line has too many statements. Maximum allowed is {{maxStatementsPerLine}}.",
-            data = { maxStatementsPerLine };
+            message = "This line has {{numberOfStatementsOnThisLine}} statement(s). Maximum allowed is {{maxStatementsPerLine}}.";
```

Comment:
I hadn't considered log parsing. However, my feeling is that if log parsing
were that important, the rule name is what should be parsed (and/or folks
can choose a formatter that is friendlier to automated parsing). To me, a
rule message should be regarded as human readable, not computer parsable,
and other fields should be used for parsing.

Let's wait for a second opinion though. :smile:

On Sep 4, 2016 6:04 PM, "Jordan Harband" <notifications@github.com> wrote:

> In lib/rules/max-statements-per-line.js
> <https://github.com/eslint/eslint/pull/7044#discussion_r77460195>:
>
> > @@ -35,11 +35,11 @@ module.exports = {
> >          const sourceCode = context.getSourceCode(),
> >              options = context.options[0] || {},
> >              maxStatementsPerLine = typeof options.max !== "undefined" ? options.max : 1,
> > -            message = "This line has too many statements. Maximum allowed is {{maxStatementsPerLine}}.",
> > -            data = { maxStatementsPerLine };
> > +            message = "This line has {{numberOfStatementsOnThisLine}} statement(s). Maximum allowed is {{maxStatementsPerLine}}.";
>
> Sure, I can do that. My thinking was that having the non-variable part be
> identical would make it easier for people doing log parsing. Should I make
> this change?
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/eslint/eslint/pull/7044/files/3a2d36721e241376db8c6e3ba89ded485706d976#r77460195>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AARWejS3tsGlv4px1lnK2kz6EdYh8RHzks5qm05-gaJpZM4J0Hsx>
> .
>